stumpy/core.py

```diff
@@ -659,17 +659,41 @@ def welford_nanstd(a, w=None):
     return np.sqrt(np.clip(welford_nanvar(a, w), a_min=0, a_max=None))
 
 
-def rolling_nanstd(a, w):
+@njit(parallel=True, fastmath={"nsz", "arcp", "contract", "afn", "reassoc"})
+def _rolling_nanstd_1d(a, w):
     """
-    Compute the rolling standard deviation for 1-D and 2-D arrays while ignoring NaNs
-    using a modified version of Welford's algorithm but is much faster than using
-    `np.nanstd` with stride tricks.
+    A Numba JIT-compiled and parallelized function for computing the rolling standard
+    deviation for 1-D array while ignoring NaN.
 
-    This a convenience wrapper around `welford_nanstd`.
+    Parameters
+    ----------
+    a : numpy.ndarray
+        The input array
 
-    This essentially replaces:
+    w : int
+        The rolling window size
+
+    Returns
+    -------
+    out : numpy.ndarray
+        This 1D array has the length of `a.shape[0]-w+1`. `out[i]`
+        contains the stddev value of `a[i : i + w]`
+    """
+    l = a.shape[0] - w + 1
+    out = np.empty(l, dtype=np.float64)
+    for i in prange(l):
+        out[i] = np.nanstd(a[i : i + w])
+
+    return out
 
-        `np.nanstd(rolling_window(T[..., start:stop], m), axis=T.ndim)`
+
+def rolling_nanstd(a, w, welford=False):
+    """
+    Compute the rolling standard deviation over the last axis of `a` while ignoring
+    NaNs.
+
+    This essentially replaces:
+        `np.nanstd(rolling_window(a, w), axis=a.ndim)`
```

Comment:
I removed `..., start:stop`

Reply:
I would keep this so that it is consistent with the other similar function

Reply:
Ultimately, this will be used in `compute_mean_std` and in there we actually use:

```
rolling_nanstd(T[..., start:stop], m)
```

Reply:
I see your point  now :) I will retrieve it.

Reply:
Should I use `T, m` (instead of `a, w`) for the sake of consistency?

Reply:
If you don't mind, can you change all of the docstrings (where applicable) to use `a` and `w` instead?

Reply:
Sure. I will revise it.